### DISCUSSION TITLE
How to switch the icon of ui.button?

### DISCUSSION THREAD

**Discussion by DUQIA** [zauberzeug/nicegui#1932] [Q&A] [answered]
### Question

```
def change_icon(event='play_arrow'):
    print(event)
    if event == 'play_arrow':
        event = 'pause'
    elif event == 'pause':
        event = 'play_arrow'
    return event
ui.button('mybutton', icon=change_icon(), on_click=lambda : change_icon('pause'))
```
I want to know how to implement this feature in nicegui.[progress-related](https://quasar.dev/vue-components/button#progress-related)
My goal is to toggle the icon of a button by clicking on it.Is it possible to merge these buttons into one [here](https://nicegui.io/documentation#audio)?

**Comment by natankeddem** [comment, marked as answer]
Like this?
```py
from nicegui import ui


def change_icon(e):
    if e.sender._props["icon"] == "pause":
        e.sender.props("icon=play_arrow")
    else:
        e.sender.props("icon=pause")


ui.button("mybutton", icon="play_arrow", on_click=change_icon)
ui.run()
```

**Reply by DUQIA** [reply]
Thank you very much for your answer, you really helped me a lot! Your solution perfectly matched my needs, and allowed me to achieve my goal.
